src/NinjaTools.FluentMockServer.API/Proxy/Visitors/Collections/QueryCollectionWrapper.cs

```diff
@@ -0,0 +1,102 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.Linq;
+using System.Net;
+using System.Web;
+using JetBrains.Annotations;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Primitives;
+using Newtonsoft.Json;
+using NinjaTools.FluentMockServer.API.Proxy.Visitors.Collections.Converters;
+using NinjaTools.FluentMockServer.API.Types;
+
+namespace NinjaTools.FluentMockServer.API.Proxy.Visitors.Collections
+{
+    /// <inheritdoc cref="IQueryCollection" />
+    [JsonConverter(typeof(QueryCollectionConverter))]
+    [JsonObject]
+    public class QueryCollectionWrapper : IQueryCollection, IVisitable, IDontRenderWhenEmpty
+    {
+        /// <inheritdoc />
+        public bool IsEnabled() => Count > 0;
+
+        [JsonIgnore]
+        public QueryCollection QueryCollection { get; set; }
+
+        [JsonIgnore]
+        public QueryString QueryString { get; set; }
+
+        [JsonIgnore]
+        public string Query => QueryString.Value;
+
+        /// <inheritdoc />
+        public void Accept(IVisitor visitor)
+        {
+            if (visitor is IVisitor<QueryCollectionWrapper> typed)
+            {
+                typed.Visit(this);
+            }
+        }
+
+        public QueryCollectionWrapper(QueryString queryString)
+        {
+            var nvc = HttpUtility.ParseQueryString(queryString.Value);
+            var querySegments = nvc.AllKeys.ToDictionary(k => k, v => new StringValues(nvc[v]));
+            QueryCollection  = new QueryCollection(querySegments);
+            QueryString = queryString;
+        }
+
+
+        /// <inheritdoc />
+        public QueryCollectionWrapper() : this(ArraySegment<KeyValuePair<string, StringValues>>.Empty)
+        { }
+
+
+        public QueryCollectionWrapper([CanBeNull] IEnumerable<KeyValuePair<string, StringValues>>? querySegments)
+        {
+            var segments =
+                (querySegments ?? throw new ArgumentNullException(nameof(querySegments)))
+                .ToList();
+
+            QueryCollection  = new QueryCollection(segments.ToDictionary(k => k.Key, v => v.Value));
+            QueryString = QueryString.Create(segments.ToList());
+        }
+
+        /// <inheritdoc />
+        public IEnumerator<KeyValuePair<string, StringValues>> GetEnumerator()
+        {
+            return QueryCollection.GetEnumerator();
+        }
+
+        /// <inheritdoc />
+        IEnumerator IEnumerable.GetEnumerator()
+        {
+            return ((IEnumerable) QueryCollection ).GetEnumerator();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Review this cast; in this project there's no type that extends 'QueryCollection' and implements 'IEnumerable'.](https://app.codacy.com/manual/git_36/NinjaTools.FluentMockServer/pullRequest?prid=4802644)